src/components/Signup/Signup.jsx

```diff
@@ -0,0 +1,255 @@
+                className={errors.userName && 'has-validation-error'}
+                onChange={this.handleChange}
+                onFocus={this.clearErrorMessages}
+                error={errors.userName}
+              />
+
+              <Input
+                type="email"
+                value={data.email}
+                name="email"
+                placeholder="Enter email"
+                className={errors.email && 'has-validation-error'}
+                onChange={this.handleChange}
+                onFocus={this.clearErrorMessages}
+                error={errors.email}
+              />
+
+              <Input
+                type="password"
+                value={data.password}
+                name="password"
+                placeholder="Enter password"
+                className={errors.password && 'has-validation-error'}
+                onChange={this.handleChange}
+                onFocus={this.clearErrorMessages}
+                error={errors.password}
+              />
+
+              <Input
+                type="password"
+                value={data.confirmPassword}
+                name="confirmPassword"
+                placeholder="Confirm password"
+                className={errors.confirmPassword && 'has-validation-error'}
+                onChange={this.handleChange}
+                onFocus={this.clearErrorMessages}
+                error={errors.confirmPassword}
+              />
+
+              {errors.signup && (
+                <div className="form-field signup-error">
+                  <span>{errors.signup}</span>
+                </div>
+              )}
+
+              <div className="form-field">
+                <Button
+                  className="btn"
+                  title="Please fill up the form"
+                  label={signupReducer.isLoading ? 'Please wait...' : 'Submit'}
+                  disabled={signupReducer.isLoading}
+                />
+              </div>
+            </Form>
+
+            <div className="Signup__form__footer">
+              <div className="social-signup">
+                Continue with:
+                <span className="social-icons">
+                  <img src={FacebookIcon} alt="Facebook Icon" />
+                </span>
+                <span className="social-icons">
+                  <img src={GoogleIcon} alt="Google Icon" />
+                </span>
+                <span className="social-icons">
+                  <img src={TwitterIcon} alt="Twitter Icon" />
+                </span>
+              </div>
+
+              <p>
+                Have an account?&nbsp;
+                <Link to="/login">Login</Link>
+              </p>
+            </div>
+          </section>
+        </main>
+      </>
+    );
+  }
+}
+
+Signup.propTypes = {
+  signupReducer: PropTypes.object.isRequired,
+  signupAction: PropTypes.func.isRequired,
+  history: PropTypes.object.isRequired,
+};
+
+const mapStateToProps = ({ signupReducer }) => ({ signupReducer });
```

Comment:
why is this called signupReducer?, is it a function that takes in an action, a state and returns a new state?

Reply:
I have changed the name to `signup`